File: JavaSdk/src/main/java/io/hackerbay/oneuptime/OneUptimeListener.java
package io.hackerbay.oneuptime;

import com.google.gson.JsonObject;
import io.hackerbay.oneuptime.model.Timeline;
import io.hackerbay.oneuptime.model.TrackerOption;

import java.util.ArrayList;

public class OneUptimeListener {
    private String currentEventId;
    private Util util;
    private OneUptimeTimelineManager oneuptimeTimelineManager;

    public OneUptimeListener(String currentEventId, TrackerOption options) {
        this.oneuptimeTimelineManager = new OneUptimeTimelineManager(options);
        this.currentEventId = currentEventId;
        this.util = new Util(options);
    }
    public void logErrorEvent(JsonObject content, String category){
        Timeline timeline = new Timeline(category, content, ErrorEventType.error.name());
        timeline.setEventId(this.currentEventId);

        this.oneuptimeTimelineManager.addItemToTimeline(timeline);
    }
    public void logCustomTimelineEvent(Timeline timelineObj) {
        timelineObj.setEventId(this.currentEventId);

        // add timeline to the stack
        this.oneuptimeTimelineManager.addItemToTimeline(timelineObj);
    }
    public ArrayList<Timeline> getTimeline() {
        // this always get the current state of the timeline array list
        return this.oneuptimeTimelineManager.getTimeline();
    }
    public void clearTimeline(String eventId) {
        // set a new eventId
        this.currentEventId = eventId;
        // this will reset the state of the timeline array list
        this.oneuptimeTimelineManager.clearTimeline();
    }
}

File: PhpSdk/src/OneUptime/OneUptimeListener.php
<?php

/**
 * @author bunday
 */

namespace OneUptime;

use stdClass;

class OneUptimeListener
{
    private $timelineObj;
    private $currentEventId;
    private $utilObj;

    /**
     * OneUptimeListener constructor.
     * @param string $eventId
     * @param array $options
     */
    public function __construct($eventId, $options)
    {
        // start the timeline manager
        $this->timelineObj = new OneUptimeTimelineManager($options);
        $this->currentEventId = $eventId;
        $this->utilObj = new Util($options);
    }
    public function logErrorEvent($content, $category = 'exception') {

        $timelineObj =  new stdClass();
        $timelineObj->category = $category;
        $timelineObj->data = $content;
        $timelineObj->type = $this->utilObj->getErrorType()->error;
        $timelineObj->eventId = $this->currentEventId;

        // add timeline to the stack
        $this->timelineObj->addToTimeline($timelineObj);
    }
    public function logCustomTimelineEvent($timelineObj) {
        $timelineObj->eventId = $this->currentEventId;

        // add timeline to the stack
        $this->timelineObj->addToTimeline($timelineObj);
    }
    public function getTimeline() {
        // this always get the current state of the timeline array
        return $this->timelineObj->getTimeline();
    }
    public function clearTimeline($eventId) {
        // set a new eventId
        $this->currentEventId = $eventId;
        // this will reset the state of the timeline array
        return $this->timelineObj->clearTimeline();
    }
}
